### PR TITLE
More hidden-definition tests

### PR DESCRIPTION
**Pull request by aisamanra**
### Motivation
We've had a few regressions related to `hidden-definition-finder` not picking up on things it used to. This expands the test suite a bit so that it'll be harder to have those regressions.

### Test plan

See included automated tests.

### CODE REVIEW

File: gems/sorbet/test/hidden-method-finder/shims.rb [hunk 1]
```diff
@@ -0,0 +1,389 @@
+# the following are included to avoid including these in
```
Comment (dmitry-stripe) [inline gems/sorbet/test/hidden-method-finder/shims.rb]:
```suggestion
# the following are included to get test output to be identical between OS X and Linux
```

Comment (DarkDimius) [reply gems/sorbet/test/hidden-method-finder/shims.rb]: (the `to avoid` description doesn't describe what does this achieve)

**Comment by aisamanra** [comment]
I think they should be reasonably stable but I'm not 100% sure: big changes to our symbol table output _might_ change them, but I don't know how often our changes in practice are going to cause them to change. They don't take long to regenerate—on the order of a minute or so, I think, but I can double-check—so they don't add too much to the overall update_exp time.

**Comment by aisamanra** [comment]
Actually, on reflection, it will need to change whenever we add things to the RBI files (as many of those things were already picked up on as hidden methods, and moving them to the RBI files makes them no longer hidden) so it might be more common than I thought. They don't take long to regenerate, though: if you already have the relevant Ruby versions built for testing, then it takes about 7 seconds to regenerate all of them.

**Pull request merged**